### PR TITLE
chore: protect v3.x.x branch

### PR DESCRIPTION
**Pull request by tswast**
In preparation for breaking changes.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Review by busunkim96 — APPROVED** [review]

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 branchProtectionRules:
 # Identifies the protection rule pattern. Name of the branch to be protected.
 # Defaults to `master`
-- pattern: master
+- pattern: '{master,v3.x.x}'
```
Comment (tseaver) [inline .github/sync-repo-settings.yaml:6]: Hmm, I'm not sure of the syntax -- are you planning that the actual branch name will be `v3.x.x`?  ISTM that `v3` would be clearer:  it is the branch from which `3.x.x` releases will be made.

Comment (tswast) [reply .github/sync-repo-settings.yaml:6]: Yes, I was planning `v3.x.x` to be the actual name. `v3` SGTM, though.

**Review by tseaver — APPROVED** [review]

File: .github/sync-repo-settings.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 branchProtectionRules:
 # Identifies the protection rule pattern. Name of the branch to be protected.
 # Defaults to `master`
-- pattern: master
+- pattern: '{master,v3}'
```
Comment (JustinBeckwith) [inline .github/sync-repo-settings.yaml:6]: Huh - I would be surprised if this works - is this in the docs somewhere?  @tmatsuo

Comment (tswast) [reply .github/sync-repo-settings.yaml:6]: I assumed it just passes through to the GitHub API.

I can merge and see what happens. :-)

**Pull request merged**